modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastViewModel.kt

```diff
@@ -208,6 +208,7 @@ class PodcastViewModel
     override fun onCleared() {
         super.onCleared()
         disposables.clear()
+        podcastAndEpisodeDetailsCoordinator.onEpisodeDetailsDismissed = null
```

Comment:
This fixes the following leak:

```
au.com.shiftyjelly.pocketcasts.podcasts.viewmodel.PodcastAndEpisodeDetailsCoordinator instance
│    Leaking: UNKNOWN
│    Retaining 1.7 MB in 23143 objects
│    ↓ PodcastAndEpisodeDetailsCoordinator.onEpisodeDetailsDismissed
│                                          ~~~~~~~~~~~~~~~~~~~~~~~~~
├─ au.com.shiftyjelly.pocketcasts.podcasts.viewmodel.PodcastViewModel$1 instance
│    Leaking: UNKNOWN
│    Retaining 1.7 MB in 23142 objects
│    Anonymous subclass of kotlin.jvm.internal.Lambda
│    ↓ PodcastViewModel$1.this$0
│                         ~~~~~~
╰→ au.com.shiftyjelly.pocketcasts.podcasts.viewmodel.PodcastViewModel instance
​     Leaking: YES (ObjectWatcher was watching this because au.com.shiftyjelly.pocketcasts.podcasts.viewmodel.
​     PodcastViewModel received ViewModel#onCleared() callback)
​     Retaining 1.7 MB in 23141 objects
```